File: lib/services/app_preferences_service.dart
import 'package:flutter/material.dart';
import 'package:shared_preferences/shared_preferences.dart';

class AppPreferencesService {
  static const String _themeModeKey = 'theme_mode';

  Future<ThemeMode> loadThemeMode() async {
    final preferences = await SharedPreferences.getInstance();
    final value = preferences.getString(_themeModeKey);

    switch (value) {
      case 'dark':
        return ThemeMode.dark;
      case 'light':
      default:
        return ThemeMode.light;
    }
  }

  Future<void> saveThemeMode(ThemeMode mode) async {
    final preferences = await SharedPreferences.getInstance();
    final value = switch (mode) {
      ThemeMode.dark => 'dark',
      ThemeMode.light || ThemeMode.system => 'light',
    };
    await preferences.setString(_themeModeKey, value);
  }
}

File: src/services/appPreferencesService.ts
import { ThemeMode } from '../types';

const THEME_MODE_KEY = 'theme_mode';

export class AppPreferencesService {
  loadThemeMode(): ThemeMode {
    try {
      const mode = localStorage.getItem(THEME_MODE_KEY);
      if (mode === 'dark' || mode === 'light') {
        return mode;
      }
      // Check system preference
      if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
        return 'dark';
      }
      return 'light';
    } catch {
      return 'light';
    }
  }

  saveThemeMode(mode: ThemeMode): void {
    try {
      localStorage.setItem(THEME_MODE_KEY, mode);
    } catch (e) {
      console.error('Failed to save theme mode:', e);
    }
  }
}

export const appPreferencesService = new AppPreferencesService();
